File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter05/RemotingMetadata/CustomerMetadata/CustomerStubs.vb
'The CustomerService class stub code
Public Class CustomerService 
   Inherits MarshalByRefObject
   
   Public Function Test() as Integer
      Throw New NotSupportedException("Method cannot be run locally")
   End Function
   
   Public Function CreateCustomer(name As String) As Customer
      Throw New NotSupportedException("Method cannot be run locally")
   End Function
 End Class

'The Customer class stub code
Public Class Customer 
   Inherits MarshalByRefObject
   
   Private mName as String
   
   Public Sub New(name As String)
      Throw New NotSupportedException("Method cannot be run locally")
   End Sub

   Public Function SayHello() As String
      Throw New NotSupportedException("Method cannot be run locally")
   End Function
End Class

File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter05/RemotingMetadata/CustomerMetadata/CustomerStubs.cs
using System;

namespace CustomerLibrary
{
   // The CustomerService class stub code
   public class CustomerService : MarshalByRefObject
   {
      // A basic test to make sure remoting settings are working
      public int Test()
      {
         throw new NotSupportedException("Method cannot be run locally");
      }

      // A factory method to return an Customer object
      public Customer CreateCustomer(string name)
      {
         throw new NotSupportedException("Method cannot be run locally");
      }
   }

   // The Customer class stub code
   public class Customer : MarshalByRefObject
   {
      public string SayHello()
      {
         throw new NotSupportedException("Method cannot be run locally");
      }
   }
}
